src/raftstore/store/worker/apply.rs

```diff
@@ -1161,3 +1181,339 @@ impl Runnable<Task> for Runner {
+            cmd.set_cmd_type(CmdType::Put);
+            if let Some(cf) = cf {
+                cmd.mut_put().set_cf(cf.to_owned());
+            }
+            cmd.mut_put().set_key(key.to_vec());
+            cmd.mut_put().set_value(value.to_vec());
+            self.req.mut_requests().push(cmd);
+            self
+        }
+
+        fn delete(self, key: &[u8]) -> EntryBuilder {
+            self.add_delete_req(None, key)
+        }
+
+        fn delete_cf(self, cf: &str, key: &[u8]) -> EntryBuilder {
+            self.add_delete_req(Some(cf), key)
+        }
+
+        fn add_delete_req(mut self, cf: Option<&str>, key: &[u8]) -> EntryBuilder {
+            let mut cmd = Request::new();
+            cmd.set_cmd_type(CmdType::Delete);
+            if let Some(cf) = cf {
+                cmd.mut_delete().set_cf(cf.to_owned());
+            }
+            cmd.mut_delete().set_key(key.to_vec());
+            self.req.mut_requests().push(cmd);
+            self
+        }
+
+        fn build(mut self) -> Entry {
+            self.entry.set_data(self.req.write_to_bytes().unwrap());
+            self.entry
+        }
+    }
+
+    #[test]
+    fn test_handle_raft_entry_normal() {
+        let (_path, db) = create_tmp_engine("test-delegate");
+        let mut reg = Registration::default();
+        reg.region.set_end_key(b"k5".to_vec());
+        reg.region.mut_region_epoch().set_version(3);
+        let mut delegate = ApplyDelegate::from_registration(db.clone(), reg);
+        let (tx, rx) = mpsc::channel();
+
+        let put_entry = EntryBuilder::new(1, 1)
+            .put(b"k1", b"v1")
+            .epoch(1, 3)
+            .capture_resp(&mut delegate, tx.clone())
+            .build();
+        let res = delegate.handle_raft_entry_normal(put_entry);
+        assert!(res.is_none());
+        let resp = rx.try_recv().unwrap();
+        assert!(!resp.get_header().has_error(), "{:?}", resp);
+        assert_eq!(resp.get_responses().len(), 1);
+        let dk_k1 = keys::data_key(b"k1");
+        assert_eq!(db.get(&dk_k1).unwrap().unwrap(), b"v1");
+        assert_eq!(delegate.applied_index_term, 1);
+        assert_eq!(delegate.apply_state.get_applied_index(), 1);
+
+        let lock_written_bytes = delegate.metrics.lock_cf_written_bytes;
+        let written_bytes = delegate.metrics.written_bytes;
+        let written_keys = delegate.metrics.written_keys;
+        let size_diff_hint = delegate.metrics.size_diff_hint;
+        let put_entry = EntryBuilder::new(2, 2).put_cf(CF_LOCK, b"k1", b"v1").epoch(1, 3).build();
+        delegate.handle_raft_entry_normal(put_entry);
+        let lock_handle = db.cf_handle(CF_LOCK).unwrap();
+        assert_eq!(db.get_cf(lock_handle, &dk_k1).unwrap().unwrap(), b"v1");
+        assert_eq!(delegate.metrics.lock_cf_written_bytes,
+                   lock_written_bytes + 5);
+        assert!(delegate.metrics.written_bytes >= written_bytes + 5);
+        assert_eq!(delegate.metrics.written_keys, written_keys + 2);
```

Comment:
KVs written are ("zk1", "v1") and (applied_state_key, applied_state). And written_bytes should be larger than the length of all key values.